extensions/import/src/dialogs/derivedColumnDialog.ts

```
@@ -0,0 +1,370 @@
+						'line-height': '22pt',
+						'margin-left': '1em',
+						'margin-top': '0em'
+					}
+				}).component();
+
+			this.clearAndAddTransformationContainerComponents(false);
+
+			const flexGrid = view.modelBuilder.flexContainer().withLayout({
+				flexFlow: 'row',
+				height: '100%',
+				width: '100%'
+			}).component();
+
+			/**
+			 * Setting height of the div based on the total viewport height after removing dialog
+			 * header and footer heights. With this the div will occupy the entire page space of the dialog.
+			 */
+			flexGrid.addItem(columnContainer, {
+				flex: '0 0 auto',
+				CSSStyles: {
+					'min-height': 'calc(100vh - 160px)'
+				}
+			});
+			flexGrid.addItem(this._transformationContainer, {
+				flex: '0 0 auto',
+				CSSStyles: {
+					'overflow': 'scroll',
+					'padding-right': '10px',
+					'width': '900px',
+					'max-height': 'calc(100vh - 160px)'
+				}
+			});
+
+			const formBuilder = view.modelBuilder.formContainer().withFormItems(
+				[
+					{
+						component: flexGrid
+					}
+				],
+				{
+					horizontal: false
+				}
+			);
+			const form = formBuilder.withLayout({ width: '100%' }).component();
+			return view.initializeModel(form);
+		});
+
+		this._dialogObject.okButton.onClick(e => {
+			this._doneEmitter.emit('done');
+		});
+
+		this._dialogObject.cancelButton.onClick(e => {
+			this._doneEmitter.emit('close');
+		});
+
+		this._dialogObject.content = [tab];
+		azdata.window.openDialog(this._dialogObject);
+		return new Promise((resolve) => {
+			this._doneEmitter.once('done', async () => {
+				try {
+					await this._provider.sendSaveTransformationRequest({
+						derivedColumnName: this.currentDerivedColumnName
+					});
+					resolve({
+						derivedColumnName: this.currentDerivedColumnName,
+						derivedColumnDataPreview: this.currentTransformation
+					});
+				} catch (e) {
+					console.log(e); // Need to have better error handling for saved transformation. However this seems to be mostly a non-issue.
```

the error is not surfaced to user and user will have no idea what happened.

---

in case of error, the resolve/reject is never called, and it will never return.